### PR TITLE
Remove non-working LevelKeyList, Variants, and Special Level Settings UI on Lesson Edit

### PR DESCRIPTION
**Pull request by dmcavoy**
While working on another PR I noticed that for the Lesson Editor when we were calling init for acitivitiesEditorRedux we were still using fake levelKeyList data. This got me wondering what it could be breaking that we were not passing in real levelKeyList data. In digging in a realized the only component on the Lesson Edit page that used levelKeyList was the LevelDetailsToken and specifically the parts of that token that are for variants and legacy level settings that would show in the scripts (skin, video key, level concept difficult, concepts). I had kept the UI for this in because I was guessing that we would want to set it up in the future but I'm worried that having it in there when its not working fully is possibly more harmful. Plus I think we might come up with different solutions for solving these special cases when we finally address them later (such as getting the old legacy settings moved into levels instead of in the script and using the add level UI to add variants instead of the current way).  

So I'm proposing with this PR that we remove these settings which are not working and add them back in the future when we decide how we want to address them. This also prevents having to send all the levelKeyList information down for the lesson edit page which could slow it down.

## Testing story

Relying mostly on existing tests to make sure I didn't break something that is necessary.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by bethanyaconnor — APPROVED** [review]
Thanks for doing this! I agree that I don't think we want to send the levelKeyList down unless we absolutely have to -- it's huge!

**Comment by davidsbailey** [comment]
👀  looking... sorry I am so far behind on reviews, but I want to take a close look at this one so please hold off on merging for now.

File: apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx [hunk 1]
```diff
@@ -93,69 +49,9 @@ class LevelTokenDetails extends Component {
...
-    addVariant: PropTypes.func.isRequired,
-    removeVariant: PropTypes.func.isRequired,
-    setActiveVariant: PropTypes.func.isRequired,
-    setLevelField: PropTypes.func.isRequired,
     setScriptLevelField: PropTypes.func.isRequired
   };
 
-  containsLegacyLevel() {
-    return this.props.scriptLevel.levels.some(level =>
-      /^blockly:/.test(this.props.levelKeyList[level.id])
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx:108]:
```suggestion
      /^blockly:/.test(level.key)
```

File: apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx [hunk 2]
```diff
@@ -206,105 +94,6 @@ class LevelTokenDetails extends Component {
...
-              )}
-              <span style={{...styles.levelFieldLabel}}>Level name:</span>
-              <LevelNameInput
-                onSelectLevel={this.handleLevelSelected.bind(
-                  this,
-                  index,
-                  level.id
-                )}
-                levelNameToIdMap={this.props.levelNameToIdMap}
-                initialLevelName={this.props.levelKeyList[level.id] || ''}
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx:292]:
```suggestion
                initialLevelName={level.key || ''}
```

File: apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx [hunk 3]
```diff
@@ -93,69 +49,9 @@ class LevelTokenDetails extends Component {
...
-      this.props.activityPosition,
-      this.props.activitySectionPosition,
-      this.props.scriptLevel.position,
-      variant,
-      levelId
-    );
-  };
-
-  handleAddVariant = () => {
-    this.props.addVariant(
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx:123]: this method would need to be changed to use the add level dialog to add a variant. I think this where "fixing forward" would get complicated.

**Review by davidsbailey — APPROVED** [review]
I'm having trouble with this because on the one hand we know we are going to need to add back almost all of this code for June, but OTOH we under such tight timelines for end-of-Dec and "January ASAP" that we don't really want to spend any "extra" time fixing this right now. However, I wanted to explore the fix forward, because it looked promising. here is what I found:

1. the levelKeyList was only truly ever "needed" for the feature where a level key could be typed into the UI, and we would autocomplete to valid level keys (LevelNameInput). for any existing LevelTokens, we should be able to get the level key from the server when we get the other data we need for that script level and its levels. 
2. today, the level key of the **active** level is currently available at this.props.scriptLevel.key . This was added in #37988 and #37969 (sorry for leaving you off these PRs Dani, they happened while you were out!). I don't think this will help you directly in this PR, but could be useful for background.
3. the keys of **inactive** level variants within a script level are **not** currently available on the client. however, they could be made available by making adding the level key here: https://github.com/code-dot-org/code-dot-org/blob/e95cb6146ef9f80dded3d1cad6931758c514ee1f/dashboard/app/models/script_level.rb#L453-L462 (see inline comments for updates to the UI to use level.key)
4. finally and sadly, in order for the Add Variants button to work today without levelKeyList, you'd need to hook it up to the AddLevelDialog. this seems like where it gets tricky.

After investigating this far, it seems like the updates to the AddLevelDialog would be the dealbreaker for "fixing forward" right now. Today the add level dialog adds new level tokens and is really an "add script level dialog". we would have to change it somewhat substantially to make it add new levels to existing script levels.

Having said all that, I think what you've got in this PR is probably best. I really don't want to delete code that we know we'll need again soon, but that actually probably is the best practice. Thanks for hearing me out, and thanks for cleaning this up! 

You can ignore the inline comments, they are just to illustrate the above points.

**Pull request merged**